### PR TITLE
remove the unstable `core::sync::atomic::ATOMIC_*_INIT` constants

### PR DESCRIPTION
**Pull request by kadiwa4**
Tracking issue: #99069

It would be weird to ever stabilise these as they are already deprecated.

### CODE REVIEW

File: library/core/tests/lib.rs [hunk 1]
```diff
@@ -93,7 +93,7 @@
 #![feature(const_option)]
 #![feature(const_option_ext)]
 #![feature(const_result)]
-#![feature(integer_atomics)]
+#![cfg_attr(target_has_atomic = "128", feature(integer_atomics))]
```
Comment (kadiwa4) [inline library/core/tests/lib.rs:96]: Note that the feature now only exists if `cfg(target_has_atomic = "128")`.

**Comment by jyn514** [comment]
ping @joshtriplett - this has been waiting on review for a while

**Pull request merged**